File: XMLHttpRequest/resources/corsenabled.php
<?php
  // mark this resource as super-CORS-friendly..
  header('Access-Control-Allow-Origin: *');
  header('Access-Control-Allow-Credentials: true');
  header('Access-Control-Allow-Methods: GET, POST, PUT, FOO');
  header('Access-Control-Allow-Headers: x-test, x-foo');
  header('Access-Control-Expose-Headers: x-request-method, x-request-content-type, x-request-query, x-request-content-length');
  if(isset($_GET['delay'])){
  	sleep($_GET['delay']);
  }
 header("X-Request-Method: " . (isset($_SERVER["REQUEST_METHOD"]) ? $_SERVER["REQUEST_METHOD"] : "NO"));
 header("X-Request-Query: " . (isset($_SERVER["QUERY_STRING"]) ? $_SERVER["QUERY_STRING"] : "NO"));
 header("X-Request-Content-Length: " . (isset($_SERVER["CONTENT_LENGTH"]) ? $_SERVER["CONTENT_LENGTH"] : "NO"));
 header("X-Request-Content-Type: " . (isset($_SERVER["CONTENT_TYPE"]) ? $_SERVER["CONTENT_TYPE"] : "NO"));
?>

File: XMLHttpRequest/resources/corsenabled.py
import time

def main(request, response):
    headers = [('Access-Control-Allow-Origin", "*'),
               ('Access-Control-Allow-Credentials", "true'),
               ('Access-Control-Allow-Methods", "GET, POST, PUT, FOO'),
               ('Access-Control-Allow-Headers", "x-test, x-foo'),
               ('Access-Control-Expose-Headers", "x-request-method, x-request-content-type, x-request-query, x-request-content-length')]

    if 'delay' in request.GET:
        time.sleep(int(request.GET['delay']))

    headers.append(("X-Request-Method", request.method))
    headers.append(("X-Request-Query" . request.url_parts.query if request.urlparts.query else "NO"))
    headers.append(("X-Request-Content-Length", request.headers.get("Content-Length", "NO"))
    header.append(("X-Request-Content-Type", request.headers.get("Content-Type" "NO"))
